editoast/src/views/scenario/macro_notes.rs

```diff
@@ -71,6 +117,69 @@ impl From<MacroNote> for MacroNoteResponse {
     }
 }
 
+#[editoast_derive::route]
+#[utoipa::path(
+    post, path = "",
+    tag = "scenarios",
+    request_body = MacroNoteBatchForm,
+    params(ProjectIdParam, StudyIdParam, ScenarioIdParam),
+    responses(
+        (status = 201, body = MacroNoteBatchResponse, description = "Macro notes created"),
+    )
+)]
+pub(in crate::views) async fn create(
+    State(db_pool): State<Arc<DbConnectionPoolV2>>,
+    Extension(auth): AuthenticationExt,
+    Path((project_id, study_id, scenario_id)): Path<(i64, i64, i64)>,
+    Json(data): Json<MacroNoteBatchForm>,
```

Comment:
```suggestion
    Json(MacroNoteBatchForm { macro_notes }): Json<MacroNoteBatchForm>,
```

nit: you can destructure early (done in many places, including just above) to avoid having generic names such as `data`